### PR TITLE
Add docs for each chart and docs generation/linting

### LINKED ISSUE #69: Add helm-docs automatic docs generation/testing
Use helm-docs to auto-generate docs for our Helm charts.

https://github.com/norwoodj/helm-docs

Verify that docs are always up-to-date in CI.

https://dev.to/jamiemagee/how-to-host-your-helm-chart-repository-on-github-3kd

### PR DESCRIPTION
**Pull request by dfarrell07**
See commit messages for details:

Closes: #69

### CODE REVIEW

**Comment by dfarrell07** [comment]
I sent a PR upstream to fix the markdownlint failure on raw URLs in the maintainers table: https://github.com/norwoodj/helm-docs/pull/102

**Comment by dfarrell07** [comment]
> I sent a PR upstream to fix the markdownlint failure on raw URLs in the maintainers table: [norwoodj/helm-docs#102](https://github.com/norwoodj/helm-docs/pull/102)

For now there's a commit in this PR to ignore the markdownlint raw URL rule.

And if anyone has an explanation for the GH Pages URL working to release/install charts but resolving 404, I'd love to include it in the commit message to ignore that URL in markdownlinkcheck: https://github.com/submariner-io/submariner-charts/pull/150/commits/b9a3f8478712b27e0e8effd5c087595d0b6d5a07

**Review by tpantelis — APPROVED** [review]

File: .github/workflows/linting.yml [hunk 1]
```diff
@@ -39,6 +39,29 @@ jobs:
...
+          curl -sL "https://github.com/norwoodj/helm-docs/releases/download/v${HELM_DOCS_VERSION}/helm-docs_${HELM_DOCS_VERSION}_Linux_x86_64.tar.gz" | tar zx
+          cd -
+          /tmp/helm-docs
+          if [[ ! -z "$(git status --porcelain)" ]]; then
+            echo "Helm docs not up-to-date:"
+            git status --porcelain
+            git diff
+            echo "Run helm-docs, as the CI does above, and commit the updated docs."
+            exit 1
+          fi
```
Comment (skitt) [inline .github/workflows/linting.yml:63]: This should go in the Makefile so that it’s easy to run locally too, but that can wait for a subsequent PR.

**Review by skitt — APPROVED** [review]

File: .github/workflows/linting.yml (hunk 1 above)
Comment (dfarrell07) [reply .github/workflows/linting.yml:63]: Good idea, thanks. I'm tracking that in the Helm epic now and will send a follow-up PR. https://github.com/submariner-io/submariner-charts/issues/138

**Comment by dfarrell07** [comment]
Sent https://github.com/submariner-io/submariner-charts/pull/153 to fix the bot not adding /ready-to-test.

File: .github/workflows/linting.yml (hunk 1 above)
Comment (dfarrell07) [reply .github/workflows/linting.yml:63]: https://github.com/submariner-io/submariner-charts/pull/155

**Pull request merged**